### PR TITLE
Add `metadata.json` for launches

### LINKED ISSUE #84: Add a `metadata.json` file in the `launch_data` directory
We need some more context about the launches not only for #76, but also in general. Here are some fields I propose which should be added for every launch:

- Temperature at that day
- Air pressure at that day
- Wind speed + direction at that day 
- Start of log buffer in the file (index)
- Approximate launch site coordinates (I prefer decimal degrees, but fine with whatever)
- Rocket mass
- Type of motor?
- Type of IMU used
- Raw and est packet frequency
- Launch rod angle and direction
- Rocket dimensions?
- etc

This will serve as the ground truth values for the flight and make it easy to load a configuration for a particular flight and then test against those changes and our mock. Thoughts @wlsanderson /@JacksonElia?

### PR DESCRIPTION
**Pull request by harshil21**
Closes #84

### CODE REVIEW

File: launch_data/metadata.json [hunk 1]
```diff
@@ -0,0 +1,99 @@
+{
+    "purple_launch.csv": {
+        "date": "2023-12-16",
```
Comment (harshil21) [inline launch_data/metadata.json:3]: need to add the time of launch as well

**Review by JacksonElia — APPROVED** [review]
This looks good but what is the point of it? Will it be used in the new UI you're working on?

**Comment by harshil21** [comment]
Yes, and anyone who has no idea about the launch will get an idea of what happened.

File: launch_data/metadata.json [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+        "imu_details": {
+            "imu_model": "Parker-LORD 3DM-CX5-15",
+            "raw_packet_frequency": 1000.0,
+            "est_packet_frequency": 500.0
+        },
+        "flight_data": {
+            "log_buffer_index": 5000,
+            "apogee_meters": 457.68
+        },
+        "flight_description": "Unknown."
```
Comment (harshil21) [inline launch_data/metadata.json:24]: @JacksonElia since you were there last year, some context would be nice

**Pull request merged**